src/System.Management.Automation/engine/hostifaces/HostUtilities.cs

```diff
@@ -863,6 +864,60 @@ internal static bool IsProcessInteractive(InvocationInfo invocationInfo)
 
         #region Public Access
 
+        #region Runspace Invoke
+
+        /// <summary>
+        /// Helper method to invoke a PSCommand on a given runspace.  This method correctly invokes the command for 
+        /// these runspace cases:
+        ///   1. Local runspace.  If the local runspace is busy it will invoke as a nested command.
+        ///   2. Remote runspace.
+        ///   3. Runspace that is stopped in the debugger at a breakpoint.
+        /// Error and information streams are ignored and only the command result output is returned.
+        /// </summary>
+        /// <param name="runspace">Runspace to invoke the command on</param>
+        /// <param name="command">Command to invoke</param>
+        /// <returns>Collection of command output result objects</returns>
+        public static Collection<PSObject> InvokeOnRunspace(PSCommand command, Runspace runspace)
+        {
+            if (command == null)
+            {
+                throw new PSArgumentNullException("command");
+            }
+
+            if (runspace == null)
+            {
+                throw new PSArgumentNullException("runspace");
+            }
+
+            if ((runspace.Debugger != null) && runspace.Debugger.InBreakpoint)
+            {
+                // Use the Debugger API to run the command when a runspace is stopped in the debugger.
+                PSDataCollection<PSObject> output = new PSDataCollection<PSObject>();
+                runspace.Debugger.ProcessCommand(
+                    command,
+                    output);
+
+                return new Collection<PSObject>(output);
+            }
+
+            // Otherwise run command directly in runspace.
+            PowerShell ps = PowerShell.Create();
+            ps.Runspace = runspace;
+            ps.IsRunspaceOwner = false;
+            if (runspace.ConnectionInfo == null)
+            {
+                // Local runspace.  Make a nested PowerShell object as needed.
+                ps.SetIsNested(runspace.GetCurrentlyRunningPipeline() != null);
+            }
```

Comment:
Yes this is a potential race condition but it is handled Ok.  If the runspace state changes before the PowerShell command is run then it fails as expected.  The assumption is that the host using this API has control of the runspace and won't run another command before running this intended one.  If commands are being run from multiple threads then there is no guarantee this PowerShell command will run successfully and is by design.